src/canvas.ts

```diff
@@ -0,0 +1,138 @@
+import { Renderer, basePoint } from './utilities/renderer';
```

Comment:
[
  {
    "start": 51,
    "end": 72,
    "message": "The use of chained optional and non-optional properties can create confusion on code readability. Instead, type check the parameter and destruct into non-optional properties or apply validation.\nReplace RectangleProperties = Omit<Line, 'end'> & Partial<Pick<Line, 'end'>> & Partial<Size>  with  interface RectangleProperties {\n  start: Point;\n  end?: Point | undefined | null | '' | 0;\n  width?: number | undefined | null | '' | 0;\n  height?: number | undefined | null | '' | 0;\n}\n"
  },
  {
    "start": 87,
    "end": 127,
    "message": "context.closePath() and context.restore() are being called everytime a drawing method is being used, but it is already being called in the class get method. Remove the one that is unnecessarily called.\nReplace all \"context.closePath();\ncontext.restore();\" to \"context.restore();\"."
  },
  {
    "start": 131,
    "end": 154,
    "message": "The variable ratio that allows modification of the size of the image seems to be redundant since the size object can be used instead. Instead, remove the ratio parameter.\nReplace interface ImageProperties {\n  image: CanvasImageSource;\n  start: Point;\n  imageStart: Point;\n  size: Size;\n  ratio?: number;\n}\nwith interface ImageProperties {\n  image: CanvasImageSource;\n  start: Point;\n  imageStart: Point;\n  size: Size;\n}\n"
  },
  {
    "start": 50,
    "end": 50,
    "message": "The import for Renderer and basePoint is not needed."
  },
  {
    "start": 80,
    "end": 82,
    "message": "The way to create a plain object is to either use destructuring or object spread. Instead of destructing the object properties and returning an object directly, return the destructured object.\nReplace this:\nconst point: Point = { x, y };\nreturn { point };\nwith\n return { point: { x, y } };\n"
  },
  {
    "start": 121,
    "end": 123,
    "message": "Method draw is a duplication of method fill. Instead, remove method draw.\nRemove method draw() in getDrawBuilder function."
  },
  {
    "start": 127,
    "end": 130,
    "message": "Returning the returned value of getDrawBuilder is redundant. Remove the returned code.\nRemove the getDrawBuilder(context) in fill() and stroke() method."
  }
]